apps/web/components/settings/platform/oauth-clients/oauth-client-form/create.tsx

```diff
@@ -0,0 +1,273 @@
+import { useRouter } from "next/navigation";
+import React, { useState, useCallback } from "react";
+import { useForm, useFieldArray } from "react-hook-form";
+
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { PERMISSIONS_GROUPED_MAP } from "@calcom/platform-constants/permissions";
+import { showToast } from "@calcom/ui";
+import { Button, TextField, Label, Tooltip } from "@calcom/ui";
+
+import { useCreateOAuthClient } from "@lib/hooks/settings/organizations/platform/oauth-clients/usePersistOAuthClient";
+
+type FormValues = {
+  name: string;
+  logo?: string;
+  permissions: number;
+  eventTypeRead: boolean;
+  eventTypeWrite: boolean;
+  bookingRead: boolean;
+  bookingWrite: boolean;
+  scheduleRead: boolean;
+  scheduleWrite: boolean;
+  appsRead: boolean;
+  appsWrite: boolean;
+  profileRead: boolean;
+  profileWrite: boolean;
+  redirectUris: {
+    uri: string;
+  }[];
+  bookingRedirectUri?: string;
+  bookingCancelRedirectUri?: string;
+  bookingRescheduleRedirectUri?: string;
+  areEmailsEnabled?: boolean;
+};
+
+export const OAuthClientForm = () => {
```

Comment:
there is a lot of duplicated code between create.tsx and edit.tsx, I would create an extra component called Fields containing the rendering part and whatever can be mutualized between create and edit